### DISCUSSION TITLE
Keyboard backlight not working properly G814JV 2023

### DISCUSSION THREAD

**Discussion by DripSweat** [seerge/g-helper#3305] [Q&A]
Good day to all.

_"If this topic has already been discussed - please let me know (I didn't find a solution here)"_

i use 0.191 version
The essence of the problem or question is very simple.

Inside the program, when setting the settings "boot" and "shutdown", the keyboard does the following:

1. When turning off - animation is running, everything is fine
2. When turning off - animation is running, and then the keyboard remains on (ligth on low level) until the moment of logging into the Windows system

![image](https://github.com/user-attachments/assets/0ad49a9d-7518-450d-a43b-9d958f980464)

The question is - what needs to be done so that only the animation works, and the keyboard does not turn on in vain at the moment of loading the operating system.

Reworking the method of launching the program g-helper (set the launch for "all", and before the user logs into the system) somewhat improved the situation, but the keyboard still remains on until the moment of starting the program.

Additional question - can the program "OpenRGB" solve these problems and will there be conflicts with g-helper?

**Comment by seerge** [comment]
@DripSweat hello,

That's not what Boot and Shutdown hardware keyboard states do.

Boot - is a simple double blink animation before device starts booting into windows. Everything after that is Awake state. 
Shutdown - is a similar double blink animation after device powers off.

**Reply by DripSweat** [reply]
ok, it became a little clearer. Thanks.
But there is another problem - if I turn off the "awake" option, the keyboard backlight will not turn on at all.

That is, I turned off "awake" - the backlight no longer works)

**Reply by seerge** [reply]
"Awake" hardware state starts as soon as system starts booting. If you turn it off - then expectedly it will stay off all the time.

**Reply by DripSweat** [reply]
It becomes even clearer how it works (I didn't read everything, apparently)
A stupid question, maybe - and the "openrgb" program will be able to solve this issue?
For some reason I thought that the keyboard has its own controller to which we tell what to do - isn't it?

**Reply by seerge** [reply]
Unless OpenRGB can run even before windows starts loading - it won't make any difference :)

**Reply by DripSweat** [reply]
Got it, then I'll think about it.
Thanks for the tips...
Although I have an idea - what if the "awake" setting is assigned to a hotkey - can't that be done?
So that turning on the backlight activates this function.

**Reply by seerge** [reply]
You can just keep Awake on and adjust backlight brightness (or set it OFF) with usual hotkeys.

**Reply by DripSweat** [reply]
good day, I checked everything. Apparently either there is a problem with the drivers or something is not working properly (or I did not understand the essence of Your thought)

What I checked:
Turned off the "Awake" mode - turned off the laptop, turned it on, the backlight does not work. Even though the "boot" options are set (as I understand it, this is normal)
Next, for testing - turned on the "Awake" option - turned off and turned on the laptop. When turned on, the keyboard began to count at maximum brightness. In the settings of the "G-helper" program - the brightness is off.
That is, in theory - with the "Awake" function turned on, the brightness is off - the backlight should not be there. But it is ...

I can't understand what needs to be done so that when I turn on the laptop, there is no keyboard backlight)

**Reply by seerge** [reply]
When Awake is off - it means that backlight of your keyboard is turned OFF completely in all stages except boot/shutdown sequences. 

If you need backlight to work - you need to check Awake ON

**Reply by DripSweat** [reply]
additionally installed the following:
After updating to version 0.197 and enabling autorun, when starting the laptop - charging stops, but the backlight will glow until you log in to the system.

Earlier, when I installed the program launch myself (I wrote about this in another thread) - the keyboard backlight turned off along with the stop of charging, until the user logged in to the system.

**Reply by DripSweat** [reply]
> You can just keep Awake on and adjust backlight brightness (or set it OFF) with usual hotkeys.

Then I got it wrong here?
If I set the "awake" option to on - and set the brightness level to zero, then the backlight should be off when the laptop boots, or not?

**Reply by seerge** [reply]
@DripSweat it should be OFF, unless your laptop has a habit to reset backlight brightens settings on reboot (similar to charge limit). Some models do ...

**Reply by DripSweat** [reply]
> @DripSweat it should be OFF, unless your laptop has a habit to reset backlight brightens settings on reboot (similar to charge limit). Some models do ...

Apparently so. Resets...
Maybe it depends on the BIOS firmware?

**Reply by seerge** [reply]
On BIOS firmware or even keyboard firmware

**Reply by DripSweat** [reply]
ok.
Then the last question on this topic - is it possible to bind the "awake" option to a hotkey (or key combination)? So that it can be turned on or off with a hotkey.

**Reply by seerge** [reply]
It's too specific, you can't do that atm
